hack/deployer/runner/tanzu.go

```diff
@@ -442,4 +443,45 @@ func (t *TanzuDriver) restoreInstallerState() error {
 		WithoutStreaming().Run()
 }
 
+func (t *TanzuDriver) Cleanup(prefix string, olderThan time.Duration) ([]string, error) {
+	if err := t.loginToAzure(); err != nil {
+		return nil, err
+	}
+	daysAgo := time.Now().Add(-olderThan)
+
+	params := map[string]interface{}{
+		"Date":                 daysAgo.Format(time.RFC3339),
+		"Location":             t.plan.Tanzu.Location,
+		"E2EClusterNamePrefix": prefix,
+	}
+
+	resourceGroupListCmd := `az group list --query "[?location=='{{.Location}}']" --query "[?contains(name,'{{.E2EClusterNamePrefix}}-tanzu')]" | jq -r '.[].name'`
+	resourceGroups, err := exec.NewCommand(resourceGroupListCmd).AsTemplate(params).WithoutStreaming().OutputList()
+	if err != nil {
+		return nil, err
+	}
+
+	var deletedClusters []string
+	for _, rg := range resourceGroups {
+		params["ResourceGroup"] = rg
+
+		clustersCmd := `az resource list -l {{.Location}} -g {{.ResourceGroup}} --resource-type "Microsoft.Compute/virtualMachines" --query "[?tags.project == 'eck-ci']" | jq -r --arg d "{{.Date}}" 'map(select((.createdTime | . <= $d) and (.name|test("{{.E2EClusterNamePrefix}}-tanzu"))))|.[].name' | grep -o '{{.E2EClusterNamePrefix}}-tanzu-[a-z]*-[0-9]*' | sort | uniq`
+		clustersToDelete, err := exec.NewCommand(clustersCmd).AsTemplate(params).WithoutStreaming().OutputList()
+		if err != nil {
+			return nil, fmt.Errorf("while running az resource list command: %w", err)
+		}
+
+		for _, cluster := range clustersToDelete {
+			t.plan.ClusterName = cluster
+			t.plan.Tanzu.ResourceGroup = cluster
+			log.Printf("deleting cluster %s\n", cluster)
+			if err = t.delete(); err != nil {
```

Comment:
Same here: this is not going to work. You need to either call `Execute` or run the `setup` function

Reply:
Yeah, this is due to my unfamiliarity with these wrapped docker commands. I've updated all of these and am testing now with existing tanzu clusters that need to be deleted, and a custom ocp cluster I can fake needing to be deleted. This will verify these cleanup commands for both of these. Thanks for catching this.